src/main/kotlin/resource/UserResource.kt

```diff
@@ -34,6 +57,76 @@ fun Route.user(userService: UserService) {
                 call.respond("You are missing one or multiple parameters")
             }
         }
+
+        webSocket("/balance") {
+            try {
+                val user = TokenManager.verifyTokenAndRetrieveUser(call.parameters)
+                val balanceManager = userService.buildBalanceManager(user)
+                val balanceIncreaseRateManager = BalanceIncreaseRateManager(user)
+
+                SessionSemaphore.acquireBalanceSession(user)
+
+                try {
+                    while (true) {
```

Comment:
while(true) ... I hope you know what you are doing here...

Reply:
Yes, it's a socket so that's fine in that case!